### DISCUSSION TITLE
Finite dose

### DISCUSSION THREAD

**Discussion by MelanieMartha26** [Open-Systems-Pharmacology/Forum#1076] [Questions & Problems] [answered]
Dear everyone, 

I was wondering whether it was possible to insert an event with a finite dose for the skin permeation model, and if so, how this can be achieved. Thank you very much in advance!!

Best regards, 

Mèlanie

**Comment by abdullahhamadeh** [comment]
Hello Mèlanie,

You may set one or more dosing events in the Events building block of arbitrary size and start time, as described in section 4.2 of the user guide: 

![image](https://user-images.githubusercontent.com/18409098/148088627-0357cffe-bcc5-43e0-9931-cdd791d33f42.png)

You can also set the volatility of the vehicle to model the four dosing scenarios in Dancik et al, 2013 (cases I-IV) using the "Volatile vehicle" and "Vehicle type" inputs in the favorites menu.  eg to model cases I/II you need to set Volatile vehicle = 1 or Vehicle type = 0 (where 0 corresponds to a neat application of the permeant to skin).

![image](https://user-images.githubusercontent.com/18409098/148089029-12a52524-67ee-4daf-ace3-347b182e7ca2.png)

Hope this helps, and please let me know if any clarification is needed.

**Comment by MelanieMartha26** [comment]
Dear @abdullahhamadeh thank you very much for your swift response. I do understand how to change an event; however, it is not yet completely clear to me how one can remove the permeant that is still left on the skin after several hours. For example, I would like to simulate what happens when you apply a dose and then after 8 hours "clean" the skin. I want to simulate the model for more than 8 hours as I am curious what the end systemic exposure will be.

I hope this is clear to you, otherwise please don't hesitate to tell me.

Best regards, 

Mèlanie

**Comment by abdullahhamadeh** [comment, marked as answer]
Hi @MelanieMartha26 

Every time a dose is "added" it is assumed that any dose that was previously on the skin surface is removed.  So what you need here can be accomplished by simply adding another **event_with_dose_mass_entered_manually** event group where the dose is zero.  

Step 1: First, save **event_with_dose_mass_entered_manually** as a PKML file.  Then delete **event_with_dose_mass_from_spatial_structure**. 

Step 2: Rename **event_with_dose_mass_entered_manually** to **dose_addition**.

![image](https://user-images.githubusercontent.com/18409098/148223158-5d6d6175-518e-4b8a-b738-e16dd825c241.png)


Step 3: Right-click on **skin_application** and select **Load Event Group...** and select the previously saved PKML file containing **event_with_dose_mass_entered_manually**.

![image](https://user-images.githubusercontent.com/18409098/148223416-17d929d6-871d-4c87-b67c-70353404534a.png)

Step 4:  Rename the newly-added Event Group to **dose_removal**.

![image](https://user-images.githubusercontent.com/18409098/148223657-c7952ef2-83f9-4d15-a876-5c5300747bdd.png)
 
Step 5: Set the start times and doses for the dose addition and dose removal.  For example, under **dose_addition**, you can add a 10 ug/sq cm dose at time = 0 hours by setting `Dose_time = 0 h` and `Mass_dose_per_area = 10 µg/cm²`.  Under **dose_removal**, you can remove the previously-added dose at 12 hours by setting `Dose_time = 12 h` and `Mass_dose_per_area = 0 µg/cm²`.

![image](https://user-images.githubusercontent.com/18409098/148223758-e71b548f-ad24-4c59-966e-d06228d9f6cf.png)


Step 6: You can verify the addition and removal works by simulation the model with this new event structure and plotting the results of the `Vehicle mass per area` observer, which shows the removal of the dose at 12 hours:

![image](https://user-images.githubusercontent.com/18409098/148224448-c0b05401-8659-4de2-a031-72bc2e07606a.png)

**Comment by MelanieMartha26** [comment]
Dear @abdullahhamadeh 

Thank you very much! I tried this earlier, but somehow I made some mistakes, now I get it, thank you!

**Comment by MelanieMartha26** [comment]
Dear @abdullahhamadeh, I have another question concerning this topic: Is it also possible to set the volume of a compartment to 0 in an event? I want to make sure that after I wash the skin after 8 hours that also the passive absorption from the SC to the vehicle stops. Now an equilibrium arises for certain compounds after 8 hours, where the test item diffuses from the SC to the vehicle outside of the skin. This, however, will not happen in real-life as you completely dry the skin. 

Do you perhaps know how this can be achieved? If I need to explain myself further, please let me know!

**Reply by abdullahhamadeh** [reply]
Hi @MelanieMartha26 

One quick way of achieving this would be to introduce a parameter (equal to 1) that multiplies the passive transport flux from the vehicle to the SC here:

![image](https://user-images.githubusercontent.com/18409098/152061277-1444ba19-568d-4a02-8bca-b3130007af41.png)

and then create an event that sets this parameter to zero at time = 8 hours.  Does that describe the process you have in mind?
